packages/core/elasticsearch/core-elasticsearch-server-internal/src/version_check/ensure_es_version.ts

```diff
@@ -164,9 +180,15 @@ export const pollEsNodesVersion = ({
         })
       );
     }),
-    map((nodesInfoResponse: NodesInfo & { nodesInfoRequestError?: Error }) =>
-      mapNodesVersionCompatibility(nodesInfoResponse, kibanaVersion, ignoreVersionMismatch)
-    ),
-    distinctUntilChanged(compareNodes) // Only emit if there are new nodes or versions or if we return an error and that error changes
+    map((nodesInfoResponse: NodesInfo & { nodesInfoRequestError?: Error }) => {
+      return mapNodesVersionCompatibility(nodesInfoResponse, kibanaVersion, ignoreVersionMismatch);
+    }),
+    // Only emit if there are new nodes or versions or if we return an error and that error changes
+    distinctUntilChanged(compareNodes),
+    tap((nodesVersionCompatibility) => {
+      if (nodesVersionCompatibility.isCompatible) {
+        isStartup$.next(false);
+      }
+    })
```

Comment:
I think it'd make sense to add a `shareReplay(1)` at the end here.
If we have multiple subscriptions we will have multiple `timer()` running in parallel, and in turn multiple calls to ES.